source/hp/fe_collection.cc

```diff
@@ -103,6 +109,69 @@ namespace hp
 
 
 
+  template <int dim, int spacedim>
+  unsigned int
+  FECollection<dim, spacedim>::find_most_face_dominating_fe_in_subset(
+    const std::set<unsigned int> &fes) const
+  {
+    const hp::FECollection<dim, spacedim> &fe_collection = *this;
+
+    // If the set of elements to be dominated contains only a single element X,
+    // then by definition the dominating set contains this single element
+    // (because each element can dominate itself).
+    // There may also be others, in which case we'll check if any of these
+    // elements is able to dominate all others. If one was found, we stop
+    // looking further and return the dominating element.
+    if (fes.size() == 1)
+      {
+        Assert(*fes.begin() < fe_collection.size(),
+               ExcIndexRangeType<unsigned int>(*fes.begin(),
+                                               0,
+                                               fe_collection.size()));
+
+        return *fes.begin();
+      }
+
+    // first loop over all FEs and check which can dominate those given in @p fes:
+    for (const auto &current_fe : fes)
+      {
+        Assert(current_fe < fe_collection.size(),
+               ExcIndexRangeType<unsigned int>(current_fe,
+                                               0,
+                                               fe_collection.size()));
+
+        FiniteElementDomination::Domination domination =
+          FiniteElementDomination::either_element_can_dominate;
+
+        for (const auto &other_fe : fes)
+          if (other_fe != current_fe)
+            {
+              Assert(other_fe < fe_collection.size(),
+                     ExcIndexRangeType<unsigned int>(other_fe,
+                                                     0,
+                                                     fe_collection.size()));
+
+              domination =
+                domination &
+                fe_collection[current_fe].compare_for_face_domination(
+                  fe_collection[other_fe]);
+            }
+
+        // see if this element is able to dominate all the other
```

Comment:
due to `either_element_can_dominate`, is it possible to construct a case where more than one FE within `@p fes` will fit the requirement?

In other words, you take the first you found, but is there only one always?

Reply:
In cases where the `FECollection` has multiple finite elements of the same kind, let's say two `FE_Q(2)` elements, we want to pick any of them. A lot of testcases rely on this feature.